fix: Bearer token capitalization in Authorization header to comply with RFC 6750

spa-raj opened the pull request:
## What does this PR change?
* Changes the Authorization header from lowercase "bearer" to capitalized "Bearer" in core/pkg/nodestats/request.go to comply with RFC 6750 standard syntax.

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Improves OAuth authentication compatibility with servers that enforce case-sensitive Bearer token validation, while maintaining backward compatibility with case-insensitive servers

## Does this PR address any GitHub or Zendesk issues?
* Closes #3383 

## How was this PR tested?
* Verified by running existing test suites for core/pkg/nodestats and modules/prometheus-source/pkg/prom packages. All tests passed successfully.

## Does this PR require changes to documentation?
* No

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
* This bug fix should be labeled "next release" as it improves RFC compliance and OAuth server compatibility without breaking existing functionality. However, I don't have access to label the PR.

spa-raj committed:
fix: Bearer token capitalization in Authorization header to comply with RFC 6750

Signed-off-by: Sparsh <sparsh.raj30@gmail.com>

ameijer commented on core/pkg/nodestats/request.go:
```diff
@@ -56,7 +56,7 @@ func (c *NodeHttpClient) makeRequest(method string, URL string, bearerToken stri
 	}
 
 	if bearerToken != "" {
-		request.Header.Add("Authorization", "bearer "+bearerToken)
```
you are awesome, sparsh! think this would be a good opportunity to mock out a nodehttp client and inspect the outgoing response for correct capitalization?

ameijer committed:
Merge branch 'develop' into bug/bearer-oauth-case-sensitive

jwitko commented:
Thanks so much for this PR this is really going to help us

spa-raj committed:
Add comprehensive tests for Bearer token Authorization header

Tests verify that the NodeHttpClient correctly formats the Authorization
header with proper "Bearer" capitalization (RFC 6750 compliant) when making
HTTP requests with bearer tokens.

Includes:
 - Mock HTTP client implementation for capturing outgoing requests
 - Happy path test for correct Bearer capitalization verification
 - Edge case tests for empty tokens and special characters
 - Tests across different HTTP methods (GET, POST, PUT, DELETE)

Signed-off-by: Sparsh <sparsh.raj30@gmail.com>

ameijer reviewed (approved):
very nice!

ameijer commented:
@jwitko this will ship in 1.118

jwitko commented:
@ameijer thank you for following up!
